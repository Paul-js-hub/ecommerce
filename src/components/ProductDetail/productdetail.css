.image {
  display: flex;
  justify-content: center;
  width: 100%;
}
.image > img {
  width: 50%;
  margin: 16px;
}

.content-container {
  margin-top: 36px;
}