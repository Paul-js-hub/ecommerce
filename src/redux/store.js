import { configureStore } from '@reduxjs/toolkit';
import productsReducer from './products/products.js';
import cartsReducer from './cart/cart.js';

const store = configureStore({
  reducer: {
    products: productsReducer,
    cart: cartsReducer
  },
  middleware: (getDefaultMiddleware) => getDefaultMiddleware({ serializableCheck: false})
});

export default store;
