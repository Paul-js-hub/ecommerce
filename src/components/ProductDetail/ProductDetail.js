import { useDispatch } from "react-redux";
import { addToCart, getCartCount } from "../../redux/cart/cart";
import { useLocation }from 'react-router-dom'
import {Button, Col, Row, Typography} from 'antd'
import './productdetail.css'

const { Title } = Typography;
const ProductDetail = () => {
    const dispatch = useDispatch();
    const { state } = useLocation();
    const { product } = state || {};
    const handleAddToCart = () => {
      dispatch(addToCart(product))
      dispatch(getCartCount());
    }
    return product ? (
        <Row>
            <Col span={12}>
            <div className="image">
            <img src={product.image} alt='product'/>
            </div>
            </Col>
            <Col span={12}>
            <div className="content-container">
            <Title level={2}>{product.title}</Title>
            <p>{product.description}</p>
            <strong><p>$ {product.price}</p></strong>
            </div>
            <Button data-testid="add-to-cart" type='primary' onClick={handleAddToCart}>Add to Cart</Button>
            </Col>
        </Row>
    ) : " No product found";
}

export default ProductDetail