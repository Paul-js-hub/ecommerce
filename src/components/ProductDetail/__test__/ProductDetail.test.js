import { render, waitFor, fireEvent, screen, cleanup } from "@testing-library/react";
import ProductDetail from "../ProductDetail";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import { HashRouter as Router } from "react-router-dom";
import store from "../../../redux/store";

describe("Test Product detail component", () => {
    afterEach(cleanup);
  test("it displays product details", async () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Router>
            <ProductDetail />
          </Router>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
