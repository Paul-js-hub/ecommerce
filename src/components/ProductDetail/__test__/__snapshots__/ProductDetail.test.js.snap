// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Product detail component it displays product details 1`] = `" No product found"`;
